### PR TITLE
Fix docs workflow failure due to Fiona package

### PR DESCRIPTION
**Pull request by Conengmo**
Closes https://github.com/python-visualization/folium/issues/2002

Add `fiona` as an explicit dev dependency.

It is needed in this example in the docs https://python-visualization.github.io/folium/latest/user_guide/geojson/geopandas_and_geo_interface.html#Use-any-object-with-__geo_interface__

The other issue is with the heatmap verification. Seems like Chrome added some UI element to the headless browser that changed the window size. So I updated the verification image to correspond with the new size.

But now the size differ between my local computer and the remove env. I guess we should either rewrite this test to not be dependent on screen size or scrap it altogether.

### CODE REVIEW

**Comment by Conengmo** [comment]
I added `fiona` as a dependency but it's not being installed in Miniconda... weird

**Comment by Conengmo** [comment]
Can't get Fiona to install from the requirements. Instead I just install it with Pip explicitely. Not the nicest solution but it does work.

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -33,6 +33,11 @@ jobs:
           --file requirements.txt
           --file requirements-dev.txt
 
+    # For some reason Fiona doesn't get installed from requirements-dev...
+    - name: Install fiona dependency
+      shell: bash -l {0}
+      run: python -m pip install fiona
```
Comment (ocefpaf) [inline .github/workflows/deploy-docs.yml:39]: If you can wait I'll take a look at this next week. Something is not right here...

Comment (Conengmo) [reply .github/workflows/deploy-docs.yml:39]: something's weird right! I can definitely wait, very interesting to figure out what's going on.

Note that https://github.com/python-visualization/folium/pull/2007 is waiting for the fix to get the docs going again. Waiting a week is not an issue, but if it's much longer then maybe a quick fix is better. Definitely not meant to rush you! More of an expectation thing.

I did also manage to fix the Selenium heatmap test issue in this PR, but I'll spin that out in a separate one and merge that earlier.

Comment (hansthen) [reply .github/workflows/deploy-docs.yml:39]: I split the underlying issues for the Selenium heatmap and this change. Once we have both fixed shall we make a new release?

Comment (Conengmo) [reply .github/workflows/deploy-docs.yml:39]: sounds good!

Comment (ocefpaf) [reply .github/workflows/deploy-docs.yml:39]: > something's weird right! I can definitely wait, very interesting to figure out what's going on.

Found it. See #2012. Basically we are limiting to what is on `main` to build/test the docs.

**Pull request closed without merging**